compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -261,9 +262,17 @@ pub struct Elaborator<'context> {
     /// be wrapped in another error that will include this reason.
     pub(crate) elaborate_reasons: im::Vector<ElaborateReason>,
 
-    /// Tracks functions that had errors during elaboration.
-    /// Used to prevent the interpreter from running functions with errors.
-    pub(crate) functions_with_errors: HashSet<FuncId>,
+    /// Tracks expressions that encountered errors during elaboration.
+    /// Used by the interpreter to skip evaluation of errored expressions.
+    pub(crate) exprs_with_errors: HashSet<ExprId>,
+
+    /// Tracks statements that encountered errors during elaboration.
+    /// Used by the interpreter to skip evaluation of errored statements.
+    pub(crate) stmts_with_errors: HashSet<StmtId>,
+
+    /// Set to true when the interpreter encounters an errored expression/statement,
+    /// causing all subsequent comptime evaluation to be skipped.
+    pub(crate) comptime_evaluation_halted: bool,
```

Comment:
I don't think we necessarily need to skip all subsequent calls, but it's something that we can easily remove in the future by removing this flag so I'm fine with merging it now and slightly changing our behavior later.

I do think this would probably be more easily put in the NodeInterner. Although if it were there, that'd change our behavior from "stop once per crate" to "stop once for all crates."